### PR TITLE
ENH: maybe_convert_objects add boolean support with NA

### PR DESCRIPTION
**Pull request by phofl**
This is necessary for read_sql and nullables.

I could not find a usage of ``convert_to_nullable_integer``, so we should be able to repurpose the keyword

### CODE REVIEW

File: pandas/_libs/lib.pyx
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:1319]: i think both this and is_bool above should check that `self.bool_` is True.  Not really needed for correctness given the way it is used, but future-proofing against mistakes

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:1319]: could also rewrite is_bool as `self.is_bool_or_na and not self.null_`

Comment (phofl) [reply pandas/_libs/lib.pyx:1315]: good point, this looks better. Changed

File: pandas/_libs/lib.pyx
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:2615]: should we default to this or the pyarrow version?

Comment (mroeschke) [reply pandas/_libs/lib.pyx:2615]: I'm fine deferring this to `'boolean'`. In my IO work context, where there's no pyarrow reader, I've been generally relying on these functions to do inference & return EA types then converting that to pyarrow after. 

Could consider changing in the future to prevent a conversion

Comment (phofl) [reply pandas/_libs/lib.pyx:2615]: Agreed, this would make us inconsistent with other areas where we are doing similar things

File: pandas/_libs/lib.pyx
Comment (mroeschke) [inline pandas/_libs/lib.pyx:2371]: Just to confirm `FloatArray` isn't supported yet?

Comment (phofl) [reply pandas/_libs/lib.pyx:2371]: This is something I raised in the sql pr as well. Currently, we only use ea, when the regular dtype can’t hold missing values. So there is no real use case for FloatArray.

it would probably make more sense to always convert when this flag is set, but this is out of scope here

Comment (mroeschke) [reply pandas/_libs/lib.pyx:2371]: Yeah. I would opt to returning `Float64` when people specify `use_nullable_dtypes=True`, especially once (hopefully) other core devs find it useful to store `NaN` & `NA`

But sure can be a follow up

**Review by mroeschke — APPROVED** [review]
LGTM merge when ready @jbrockmendel

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @phofl

**Pull request merged**
